backend/datasets/delete.py

```diff
@@ -24,12 +26,23 @@ def delete_dataset(payload: JsonObject) -> JsonObject:
     datasets_model_class = datasets_model_with_meta()
 
     # get dataset to delete
+    dataset_id = req_body["id"]
     try:
-        dataset = datasets_model_class.get(
-            hash_key=f"DATASET#{req_body['id']}", consistent_read=True
-        )
+        dataset = datasets_model_class.get(hash_key=f"DATASET#{dataset_id}", consistent_read=True)
     except DoesNotExist:
-        return error_response(HTTPStatus.NOT_FOUND, f"dataset '{req_body['id']}' does not exist")
+        return error_response(HTTPStatus.NOT_FOUND, f"dataset '{dataset_id}' does not exist")
+
+    # Verify that the dataset is empty
+    s3_client = boto3.client("s3")
```

Comment:
not major but in other places we've initialised boto3 clients outside of the lambda handler.